internal/app/triggermessageprocessor.go

```diff
@@ -65,8 +69,33 @@ func (b *simpleTriggerServiceBinding) Config() *common.ConfigurationStruct {
 
 // triggerMessageProcessor wraps the ServiceBinding interface so that we can attach methods
 type triggerMessageProcessor struct {
-	bnd              trigger.ServiceBinding
-	messagesReceived gometrics.Counter
+	serviceBinding          trigger.ServiceBinding
+	messagesReceived        gometrics.Counter
+	invalidMessagesReceived gometrics.Counter
+}
+
+func NewTriggerMessageProcessor(bnd trigger.ServiceBinding, metricsManager bootstrapInterfaces.MetricsManager) *triggerMessageProcessor {
+	lc := bnd.LoggingClient()
+
+	mp := &triggerMessageProcessor{
+		serviceBinding:          bnd,
+		messagesReceived:        gometrics.NewCounter(),
+		invalidMessagesReceived: gometrics.NewCounter(),
+	}
+
+	if err := metricsManager.Register(internal.MessagesReceivedName, mp.messagesReceived, nil); err != nil {
+		lc.Warnf("%s metric failed to register and will not be reported: %s", internal.MessagesReceivedName, err.Error())
+	} else {
+		lc.Infof("%s metric has been registered and will be reported", internal.MessagesReceivedName)
+	}
+
+	if err := metricsManager.Register(internal.InvalidMessagesReceivedName, mp.invalidMessagesReceived, nil); err != nil {
+		lc.Warnf("%s metric failed to register and will not be reported: %s", internal.InvalidMessagesReceivedName, err.Error())
+	} else {
+		lc.Infof("%s metric has been registered and will be reported (if enabled)", internal.InvalidMessagesReceivedName)
```

Comment:
```suggestion
		lc.Infof("%s metric has been registered and will be reported", internal.InvalidMessagesReceivedName)
```

Reply:
Metrics are only reported when enabled in the configuration, so message is correct.

Reply:
ah, okay. That works :)